File: SDKs/configcat.com/v1/python/src/sdk/models/operations/update_config.py
from dataclasses import dataclass, field
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from sdk.models import shared


@dataclass
class UpdateConfigPathParams:
    config_id: str = field(metadata={'path_param': { 'field_name': 'configId', 'style': 'simple', 'explode': False }})
    

@dataclass
class UpdateConfigRequests:
    update_config_request: Optional[shared.UpdateConfigRequest] = field(default=None, metadata={'request': { 'media_type': 'application/*+json' }})
    update_config_request1: Optional[shared.UpdateConfigRequest] = field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    update_config_request2: Optional[shared.UpdateConfigRequest] = field(default=None, metadata={'request': { 'media_type': 'text/json' }})
    

@dataclass
class UpdateConfigRequest:
    path_params: UpdateConfigPathParams = field()
    request: UpdateConfigRequests = field()
    

@dataclass
class UpdateConfigResponse:
    content_type: str = field()
    status_code: int = field()
    config_model: Optional[shared.ConfigModel] = field(default=None)
    config_model_haljson: Optional[shared.ConfigModelHaljson] = field(default=None)
    

File: SDKs/configcat.com/v1/typescript/src/sdk/models/operations/updateconfig.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class UpdateConfigPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=configId" })
  configId: string;
}


export class UpdateConfigRequests extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "request, media_type=application/*+json" })
  updateConfigRequest?: shared.UpdateConfigRequest;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  updateConfigRequest1?: shared.UpdateConfigRequest;

  @SpeakeasyMetadata({ data: "request, media_type=text/json" })
  updateConfigRequest2?: shared.UpdateConfigRequest;
}


export class UpdateConfigRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: UpdateConfigPathParams;

  @SpeakeasyMetadata()
  request: UpdateConfigRequests;
}


export class UpdateConfigResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  configModel?: shared.ConfigModel;

  @SpeakeasyMetadata()
  configModelHaljson?: shared.ConfigModelHaljson;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
